src/resourceschemehandler.cc

```diff
@@ -7,26 +7,34 @@ ResourceSchemeHandler::ResourceSchemeHandler( ArticleNetworkAccessManager & arti
 }
 void ResourceSchemeHandler::requestStarted( QWebEngineUrlRequestJob * requestJob )
 {
-  QUrl url = requestJob->requestUrl();
-
-  QNetworkRequest request;
-  request.setUrl( url );
-  request.setAttribute( QNetworkRequest::CacheLoadControlAttribute, QNetworkRequest::PreferCache );
-  QNetworkReply * reply = this->mManager.getArticleReply( request );
-  connect( reply, &QNetworkReply::finished, requestJob, [ = ]() {
-    if ( reply->error() == QNetworkReply::ContentNotFoundError ) {
+  const QUrl url = requestJob->requestUrl();
+  QString content_type;
+  const QMimeType mineType                    = db.mimeTypeForUrl( url );
+  const sptr< Dictionary::DataRequest > reply = this->mManager.getResource( url, content_type );
+  content_type                                = mineType.name();
+  if ( reply->isFinished() ) {
+    emit reply->finished();
+  }
+  connect( reply.get(), &Dictionary::DataRequest::finished, requestJob, [ = ]() {
+    if ( !reply.get() ) {
       requestJob->fail( QWebEngineUrlRequestJob::UrlNotFound );
       return;
     }
-    if ( reply->error() != QNetworkReply::NoError ) {
-      qDebug() << "resource handler failed:" << reply->error() << ":" << reply->request().url();
-      requestJob->fail( QWebEngineUrlRequestJob::RequestFailed );
+    const auto & data = reply->getFullData();
+    if ( data.empty() ) {
+      requestJob->fail( QWebEngineUrlRequestJob::UrlNotFound );
       return;
     }
-    QMimeType mineType  = db.mimeTypeForUrl( url );
-    QString contentType = mineType.name();
+    QByteArray * ba  = new QByteArray( data.data(), data.size() );
+    QBuffer * buffer = new QBuffer( ba );
```

Comment:
the `buffer` should be delete when `requestJob` deleted.

How to delete the `ba` ？

Reply:
@shenlebantongying can you help to check the above issue?

Reply:
```
  connect( requestJob,
           &QObject::destroyed,
           buffer,
           [&ba, &buffer]() {
             ba->clear();
             delete ba;
             delete buffer;
           } );
```